lib/services/access_request_test.go

```diff
@@ -1775,30 +1776,34 @@ func TestGetRequestableRoles(t *testing.T) {
 	user := g.user(t)
 
 	tests := []struct {
-		name               string
-		userRole           string
-		requestedResources []types.ResourceID
-		disableFilter      bool
-		allowedResourceIDs []types.ResourceID
-		expectedRoles      []string
+		name                                string
```

Comment:
Test changes:

- rename `can view resource but not assume role` → `can request resource but not assume role`
- add `confirm can request resource when filtering disabled`
- don’t know how I managed to reorder `filtering disabled` with `filter by resources` - sorry for the unnecessary nosie

Adding `expectedApplicableRolesForResources` made it easier for me to comprehend what's going on here and how those two thing relate, but it may be subjective. I won't disagree with that.

Naming PR `Better ...` may be controversial. It was easier to understand it for me. I hoped it may be the same for others. If the general consensus is it makes less understandable then I'm happy to close it.